common/src/main/java/com/sequenceiq/cloudbreak/quartz/metric/JobMetricsListener.java

```diff
@@ -46,7 +53,28 @@ public void jobWasExecuted(JobExecutionContext context, JobExecutionException jo
                 jobException == null ? "without exception" : jobException.getMessage());
         QuartzMetricType metricType = jobException == null ? QuartzMetricType.JOB_FINISHED : QuartzMetricType.JOB_FAILED;
         Duration jobRunTime = Duration.ofMillis(context.getJobRunTime());
+        logJobDurationIfCritical(context);
         metricService.recordTimerMetric(metricType, jobRunTime,
                 JOB_GROUP.name(), jobKey.getGroup());
     }
+
+    private void logJobDurationIfCritical(JobExecutionContext context) {
+        long jobRunTime = context.getJobRunTime();
+        if (jobRunTime > MILLI_SECONDS_5) {
+            String category;
+            if (jobRunTime > MILLI_SECONDS_30) {
+                category = ">30s";
+            } else if (jobRunTime > MILLI_SECONDS_15) {
+                category = "15s-30s";
+            } else {
+                category = "5s-15s";
+            }
```

Comment:
Please refactor category calculation at least into a separate method or even into a separate class/enum.

The constants name is `MILLI_SECONDS_30` but the category name is `>30s` it confuses me a little.

Reply:
MILLI_SECONDS_30 is a constant used for the condition. It has nothing to do with `>30s` except the value. I don't think it worth adding more complexity to it like an enum or class or any kind of generally usable rocket that you need to click through if you want to understand it. One more method might worth, I add it then.